core/pkg/server/handler.go

```diff
@@ -416,7 +415,10 @@ func (h *Handler) handleRequestDefer(record *service.Record, request *service.De
 		// after the run has exited
 		h.systemMonitor.Stop()
 	case service.DeferRequest_FLUSH_PARTIAL_HISTORY:
-		h.activeHistory.Flush()
+		h.handleRequestPartialHistory(
```

Comment:
can you explain this?

Reply:
ah good point, i'll add a comment, it's kinda implicit in the code i guess.
But conceptually, if there is still some data in runHistory an empty partial history won't change it 
but because we force it to flush (need to add this into the partial history, it's not correct now)
it will just call handle the history with all the data that currently in the runHistory